pkg/api/pod/util.go

```diff
@@ -379,6 +379,10 @@ func dropDisabledFields(
 
 	dropDisabledCSIVolumeSourceAlphaFields(podSpec, oldPodSpec)
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.EvenPodsSpread) && !topologySpreadConstraintsInUse(oldPodSpec) {
+		// Set TopologySpreadConstraints to nil only if feature is disabled and it is not used
+		podSpec.TopologySpreadConstraints = nil
```

Comment:
This doesn't make a lot of sense to me... what is the purpose of setting it to nil if it is not used? Wouldn't that mean it is already nil?

This field is not mutable, which means on a PUT / PATCH request, it cannot be changed at all. This should therefore only apply to POST requests.

Will this get called for deployments, Replica sets, etc, templates?

Reply:
If the feature is disabled, but user still posts a payload with spreadConstraints set, it's set to nil. So basically it's for accidental prevention. I checked the other logic in `dropDisabledFields` function, they're almost using the same pattern. 

> Will this get called for deployments, Replica sets, etc, templates?

Yes, for both Pod and PodTemplate.

Reply:
i think there was a downgrade/upgrade angle to it as well, i remember someone explaining this, but i have forgotten. I agree whenever i see that piece of code , it doesnt make sense but its a pattern being followed everywhere. I think @liggitt had an explanation

Reply:
I think you misunderstood my question. the line of code has `!topologySpreadConstraintsInUse(oldPodSpec)`; it seems pointless to set it to nil if it is already not in use. Shouldn't this unconditionally set it to nil?

Reply:
My apologies... I basically agree. I will update it unless hearing from @liggitt about the potential reason.

Reply:
this sets it to nil in the *new* object if the old object did not use the field

Reply:
> Shouldn't this unconditionally set it to nil?

if the feature is enabled, you should be able to update a deployment that does not use this field to start using it

Reply:
Thanks @liggitt.

@lavalamp as Jordan pointed out, this is necessary for k8s version upgrade, or feature gate flipping within the same k8s version. So I will leave the logic as is.

Reply:
Sorry for the noise, I misread and didn't notice that oldPodSpec and podSpec were two different things.